src/client/java/dev/creesch/model/WebsocketJsonMessage.java

```diff
@@ -0,0 +1,53 @@
+package dev.creesch.model;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.google.gson.annotations.SerializedName;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.AllArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class WebsocketJsonMessage {
+    private long timestamp;
+    private ChatServerInfo server;
+    private MessageType type;
+    @SerializedName("minecraftVersion")
+    private String minecraftVersion;
+    private Object payload;
+
+
+    @Data
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class ChatServerInfo {
+        private String name;
+        private String identifier;
+    }
+
+
+    public enum MessageType {
+        @SerializedName("chatMessage")
+        CHAT_MESSAGE
+    }
+
+    // Private constructor to force use of factory methods
+    private WebsocketJsonMessage(long timestamp, ChatServerInfo server, MessageType type, Object payload, String minecraftVersion) {
+        this.timestamp = timestamp;
+        this.server = server;
+        this.type = type;
+        this.payload = payload;
+        this.minecraftVersion = minecraftVersion;
+    }
+
+    public static WebsocketJsonMessage createChatMessage(long timestamp,
```

Comment:
Style-wise I've always vastly preferred just breaking and putting params on a new line with one level of indentation.